model/source_http.go

```diff
@@ -14,7 +25,46 @@ type SourceCreateRequest struct {
 	AppCreationWorkflow string  `json:"app_creation_workflow"`
 	AvailabilityStatus  string  `json:"availability_status"`
 
-	SourceTypeID *int64 `json:"source_type_id"`
+	SourceTypeID    *int64      `json:"-"`
+	SourceTypeIDRaw interface{} `json:"source_type_id"`
+}
+
+// Validate validates that the required fields of the SourceCreateRequest request hold proper values. In the specific
+// case of the UUID, if an empty or nil one is provided, a new random UUID is generated and appended to the request.
+func (req *SourceCreateRequest) Validate() error {
+	if req.Name == nil || *req.Name == "" {
+		return fmt.Errorf("name cannot be empty")
+	}
+
+	// If no valid UUID is provided in the request, we generate one
+	if req.Uid == nil || *req.Uid == "" {
```

Comment:
My bad, the UUID should be automatically generated[1], because the spec doesn't say anything about accepting an `uuid` key in the request body.

\[1\]: https://github.com/RedHatInsights/sources-api/blob/d1d7d5dfefab699fe30853f82ec0a3d401ad27dd/app/models/source.rb#L8